source/layouts/application.erb

```diff
@@ -12,11 +12,15 @@
     <meta name="description" content="<%= page_data(:meta_description) %>">
     <meta name="keywords" content="<%= page_data(:meta_keywords) %>">
     <%= favicon_tag "favicon.ico" %>
-    <meta name="twitter:card" content="summary_large_image" />
-    <meta name="twitter:site" content="@Ubxd">
-    <meta name="twitter:title" content="<%= current_page.data.title %>">
-    <meta name="twitter:description" content="<%= truncate(strip_tags(current_page.data.header ? current_page.data.header.introduction : ""), length: 150) %>">
-    <meta name="twitter:image" content="<%= defined?(current_page.data.header.image) ? image_url("#{current_page.data.header.image}") : image_url("assets/images/unboxed-default-twitter-card.jpg") %>">
+    <% if content_for?(:twitter_card) %>
+      <%= yield_content(:twitter_card) %>
+    <% else %>
+      <meta name="twitter:card" content="summary_large_image" />
+      <meta name="twitter:site" content="@Ubxd">
+      <meta name="twitter:title" content="<%= current_page.data.title %>">
+      <meta name="twitter:description" content="<%= truncate(strip_tags(current_page.data.header ? current_page.data.header.introduction : ""), length: 150) %>">
+      <meta name="twitter:image" content="<%= defined?(current_page.data.header.image) ? image_url("#{current_page.data.header.image}") : image_url("assets/images/unboxed-default-twitter-card.jpg") %>">
+    <% end %>
```

Comment:
It looks like we only need to deal with different `twitter:description` and `twitter:image` between site-wide and blog articles.

Could we keep the `twitter:card` `twitter:site` and `twitter:title` here and then only show the other two if defined in the custom layout?

Reply:
To do it properly, we should probably make the twitter:image come from the same variable ... there's no real reason why it should be called `header.image` in some parts of the site and `main_image` in others

Reply:
In which case we could get rid of the `<% if content_for?(:twitter_card) %>` clause entirely

Reply:
Following our discussion it sounds like we're limited by the configuration in `prose.yml` - edits are made in the CMS (prose / Netlify CMS) but these can't easily be made into nested data to match other pages on the site.

I suggest we keep the top three tags here and use the custom logic for the last two. And add a comment - something like
```
<%# Most page types have a consistent header structure and rules for these, but some pages (e.g. blog posts) differ. This seems to have come about because prose doesn't allow nested YML easily. %>
```

Reply:
Ok, that makes sense ... I'll make the changes